src/main/java/net/sf/jabref/model/entry/InternalBibtexFields.java

```diff
@@ -305,17 +273,29 @@ private InternalBibtexFields(boolean serializeSpecialFields) {
             add(field);
         }
 
-        // collect all public fields for the PUBLIC_FIELDS array
-        for (BibtexSingleField sField : fieldSet.values()) {
-            if (!sField.isPrivate()) {
-                PUBLIC_FIELDS.add(sField.getFieldName());
-                // or export the complete BibtexSingleField ?
-                // BibtexSingleField.toString() { return fieldname ; }
-            }
-        }
+    }
 
-        // sort the entries
-        Collections.sort(PUBLIC_FIELDS);
+
+    public static void updateTimeStampField(String timeStampFieldName) {
```

Comment:
I think we need a thorough discussion on statics vs Globals vs injection. There are 25 imports of InternalBibtexFields in the current source (excluding model.entry), so we would have to inject in, say, half of them if we go for the Globals vs injection, plus passing through.

I do not really see it as work for this PR, although I do not deem it unlikely that it will happen eventually. (The problem is that as soon as someone starts touching these "bad" designs, one is expected to go all the way and make a perfect redesign, which rarely happens, better to iteratively improve it. With this PR it is most likely slightly simpler to modify it later and we can enforce more architecture constraints.)